leshan-tl-javacoap-server/src/main/java/org/eclipse/leshan/transport/javacoap/server/request/LwM2mResponseBuilder.java

```diff
@@ -117,14 +118,38 @@ public LwM2mResponseBuilder(CoapResponse coapResponse, CoapRequest coapRequest,
 
     @Override
     public void visit(ReadRequest request) {
+        ContentFormat contentFormat = coapResponse.options().getContentFormat() != null
+                ? ContentFormat.fromCode(coapResponse.options().getContentFormat())
+                : request.getContentFormat();
+
+        TimestampedLwM2mNode timestampedNode = null;
+        try {
+            List<TimestampedLwM2mNode> timestampedNodes = decoder.decodeTimestampedData(
+                    coapResponse.getPayload().getBytes(), contentFormat, request.getPath(), model);
+            timestampedNode = timestampedNodes != null ? timestampedNodes.get(0) : null;
+        } catch (CodecException e) {
+            LOG.debug(String.format("Unable to decode response payload of request [%s]", request));
+        }
+
         if (coapResponse.getCode().getHttpCode() >= 400) {
             // handle error response:
-            lwM2mresponse = new ReadResponse(toLwM2mResponseCode(coapResponse.getCode()), null,
-                    coapResponse.getPayloadString(), coapResponse);
+            if (timestampedNode != null && timestampedNode.isTimestamped()) {
+                lwM2mresponse = new ReadResponse(toLwM2mResponseCode(coapResponse.getCode()), null, timestampedNode,
+                        coapResponse.getPayloadString(), coapResponse);
+            } else {
+                lwM2mresponse = new ReadResponse(toLwM2mResponseCode(coapResponse.getCode()), null, null,
+                        coapResponse.getPayloadString(), coapResponse);
+            }
```

Comment:
the comment seems to says this is about error response why trying to decode timestamped node here ? :thinking: